OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs

```diff
@@ -20,24 +22,26 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("Ability")]
 	public class CaptureProperties : ScriptActorProperties
 	{
-		readonly CapturesInfo normalInfo;
+		readonly Captures[] capturesTraits;
 		readonly ExternalCapturesInfo externalInfo;
 
 		public CaptureProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			normalInfo = Self.Info.TraitInfoOrDefault<CapturesInfo>();
+			capturesTraits = Self.TraitsImplementing<Captures>().ToArray();
 			externalInfo = Self.Info.TraitInfoOrDefault<ExternalCapturesInfo>();
 		}
 
 		[Desc("Captures the target actor.")]
 		public void Capture(Actor target)
 		{
+			var captures = capturesTraits.FirstOrDefault(x => !x.IsTraitDisabled && x.Info.CaptureTypes.Overlaps(target.GetEnabledTargetTypes()));
```

Comment:
Thanks for the code, i was sure that something was wrong there. I'll apply it when i back home, i'm at hospital again.